db/db.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * The purpose of this file is to define a ste of functions that will act as an interface between the application and mongodb.
+ * The mongodb server is configured in db/config.js
+*/
+
+var User = require('./config').User;
+var Promise = require('bluebird');
+
+exports.db = {
+  getUsers: function(username, email) {
+    // Creates object to query database
+    var dbQueryObject = {};
+    if (username) {
+      dbQueryObject.username = username;
+    }
+    if (email) {
+      dbQueryObject.email = email;
+    }
+
+    return new Promise(function(resolve, reject) {
+      User.find(dbQueryObject, function(error, users) {
+        if (error) {
+          console.log('ERROR calling getUsers function', error);
+          reject(error);
+        } else {
+          resolve(users);
+        }
+      });
+    });
+  },
+
+  addUser: function(username, email, funfact, profileimage) {
+    var newDbEntry = {
+      username: username,
+      email: email,
+      funfact: funfact,
+      profileimage: profileimage
+    };
+
+    var  newUser = new User(newDbEntry);
+
+    return new Promise(function(resolve, reject) {
+      newUser.save(function(error, newUser) {
+        if (error) {
+          console.log('ERROR calling addUser function', error);
+          reject(error);
+        } else {
+          resolve(newUser);
+        }
+      });
+    });
+  },
+
+  getAllUsers: function() {
+    return new Promise(function(resolve, reject) {
+      User.find(function(error, users) {
```

Comment:
maybe: 
User.find().exec().then(function(users){//do stuff})
OR
User.find({}, function(error, users){//do stuff})

(unless your way already works in which case disregard)
http://mongoosejs.com/docs/api.html#model_Model.find